library/VersionInfoFactory.cpp

```diff
@@ -52,33 +52,28 @@ VersionInfoFactory::~VersionInfoFactory()
 
 void VersionInfoFactory::clear()
 {
-    // for each stored version, delete
-    for(size_t i = 0; i < versions.size();i++)
-    {
-        delete versions[i];
-    }
     versions.clear();
     error = false;
 }
 
-VersionInfo * VersionInfoFactory::getVersionInfoByMD5(string hash)
+std::shared_ptr<const VersionInfo> VersionInfoFactory::getVersionInfoByMD5(string hash) const
 {
-    for(size_t i = 0; i < versions.size();i++)
+    for (const auto& version : versions)
     {
-        if(versions[i]->hasMD5(hash))
-            return versions[i];
+        if(version->hasMD5(hash))
+            return version;
     }
-    return 0;
+    return nullptr;
```

Comment:
+1 here as well - we used to not be able to use `nullptr` because of GCC 4.5, but some plugins defined their own versions of it, which turned into a mess.